public/fetch.js

```diff
@@ -0,0 +1,21 @@
+const fetchURL = (method, url, requestBody, onSuccessCallback, onErrorCallback) => {
```

Comment:
`onSuccessCallback` `onErrorCallback` good idea, but you can wrap in one callback(first error callback)
and you can use it like this:

```js
if(xhr.status >= 200 && xhr.status < 300){
    cb(null, JSON.parse(xhr.responseText)) // we don't have an error so we pass null for the first argument since the error comes first
} else {
    cb(JSON.parse(xhr.responseText)) // here first argument is the error
}
```